### DISCUSSION TITLE
'Download CSV' in Kibana doesn't match displayed table columns

### DISCUSSION THREAD

**Discussion by TFWol** [Security-Onion-Solutions/securityonion#7909] [Unsupported Versions] [answered]
I don't feel like I'm doing anything wrong, but when I click my way to a nice table and press **Download CSV** on it, I'm not getting the same thing I see.

I searched the maze that is the Kibana forums and tried saving the dashboard as suggested, but that appears to not work either.

Here's a screenshot of what I see from some sample data and the related POST in Firefox.
![image](https://user-images.githubusercontent.com/9045213/167071561-20d149fc-32bc-4806-9bc4-cc4415178c98.png)

Hopefully I'm simply missing something obvious.

edit--
Good old copy-paste fails me too.
![image](https://user-images.githubusercontent.com/9045213/167072151-a322bcaf-979a-4cc1-af4f-aa2ac36f7286.png)

**Comment by dougburks** [comment, marked as answer]
I believe Kibana is designed to export the columns from the saved dashboard layout and not any temporary changes you have made. You might try creating your own custom dashboard with your custom fields. If you have further questions about Kibana's `Download CSV` feature, you should probably ask the Kibana team.

You might also want to think about the end goal for downloading to CSV. Most of our users prefer to stay in Kibana or our Hunt interface without resorting to spreadsheets.

**Reply by TFWol** [reply]
The interesting part is I had made and saved a dashboard view, but the export still didn't match up.

> You might also want to think about the end goal for downloading to CSV. Most of our users prefer to stay in Kibana or our Hunt interface without resorting to spreadsheets.

The end goal was for me to see columns arranged in a particular order in certain views to make up for the difficulty I was having with Kibana. The CSV export was just an attempt for a workaround as I've found Excel's Power Pivot works quite well for ingesting data exports.

**Reply by dougburks** [reply]
You might want to check out our new Dashboards app coming in the next release as it makes it really easy to slice and dice different fields without having to fire up Excel:
https://blog.securityonion.net/2022/05/sneak-peek-security-onion-23130-and-new.html

**Reply by TFWol** [reply]
That looks pretty slick.
It might even be just what I'm looking for as far as easy customization goes.

Thanks for answering my (at times, naive) questions.
